database.py

```diff
@@ -8,15 +9,109 @@
 
 # GitHub API datetime-stamps are already in UTC / Zulu time (+0000)
 def parse_datetime(datetime_string):
-    return datetime.datetime.strptime(datetime_string, '%Y-%m-%dT%H:%M:%SZ')
+  return datetime.datetime.strptime(datetime_string, '%Y-%m-%dT%H:%M:%SZ')
 
 
-def GetIssues(predicate=None):
-    issues = json.load(open(all_issues_file))
-    if predicate:
-        return filter(predicate, issues)
-    return issues
+def get_issues(predicate=None):
+  issues = json.load(open(all_issues_file))
+  if predicate:
+    return filter(predicate, issues)
+  return issues
 
 
-def UpdateTime(issue):
-    return parse_datetime(issue['updated_at'])
+#
+# issue helpers
+#
+
+def update_time(issue):
+  return parse_datetime(issue['updated_at'])
+
+
+def created_by(issue):
+  """Returns the user of the issue creator."""
+  return User(issue.get('user'))
+
+
+class User(object):
+
+   def __init__(self, json):
+     self.data = json
+
+   @property
+   def name(self):
+     return self.data['login']
+
+   @property
+   def link(self):
+     return self.data['html_url']
+
+
+#
+# Label helpers
+#
+
+PRIMARY_LABEL_DB = 'labels.bazel.json'
+
+Label = collections.namedtuple('Label', ['key', 'name', 'color', 'url'])
+
+class LabelDB(object):
+  """LabelsDB implements a collection of labels.
+
+  The intent is that the color choices for main repo win over those in the
+  secondary ones. To accomplish that we preload the labels from the main
```

Comment:
I'm curious - is your intention to sync labels across repositories? This is the first time I'm hearing about this idea. 

The requirements of bazelbuild/bazel is significantly different from the smaller repos. e.g. I don't have any intention for rules_jvm_external to adopt the triage policy of bazelbuild/bazel, simply because the volume is low enough that we can pay attention to every open issue.

Reply:
I do want to have tools which let me uniformly push specific labels to a large set of repos. That will help us have priorities and "hotlist" tags aligned. 

But, that is not what this change is about. This label db is purely so that if a label is in 2 repos, we favor the color from the Bazel repo. That is, it is only for making the UI better.

Reply:
I see, thanks for the clarification.